doc/developer_guide/psy_data.rst

```diff
@@ -0,0 +1,537 @@
+NVIDIA profiling tools). This class name is used as a prefix for
+the module name, the ``PSyDataType`` and functions. So if a profiling application
+is linked the above code will actually look like this::
+
+    USE profile_psy_data_mod, ONLY: profile_PSyDataType
+    TYPE(profile_PSyDataType), target, save :: profile_psy_data
+
+    CALL profile_psy_data%PreStart("update_field_mod", "update_field_code", 1, 1)
+
+.. note::
+    While adding the class prefix to the name of the instance variable
+    is not necessary, it helps improves the readability of the created code.
+
+
+The list of valid classes is specified in the configuration file. It can be
+extended by the user to support additional classes::
+
+    [DEFAULT]
+    ...
+    VALID_PSY_DATA_PREFIXES = profile extract
+
+The classes supported at the moment are:
+
+.. tabularcolumns:: |l|L|
+
+======================= =======================================================
+Class Prefix            Description
+======================= =======================================================
+profile                 All libraries related to profiling tools like DrHook,
+                        NVIDIA's profiling tools etc. See
+                        :ref:`user_guide:profiling` for details.
+extract                 For libraries that allow kernel data extraction. See
+                        :ref:`user_guide:psyke` for details.
+======================= =======================================================
+
+In the following documentation the string ``PREFIX_`` is used
+to indicate the class-prefix used (e.g. ``profile`` or ``extract``).
+
+.. note:: 
+    The transformations for profiling or kernel extraction allow
+    to overwrite the default-prefix (see 
+    :ref:`psy_data_parameters_to_constructor`).
+    This can be used to link with two different libraries of the
+    same class at the same time, e.g. you could use ``drhook_profile``
+    and ``nvidia_profile`` as class prefix. Though this also requires
+    that the corresponding wrapper libraries are modified to use this
+    new prefix.
+
+
+
+Full Example
+------------
+The following example shows the full code created by PSyclone for an 
+kernel extraction (excluding declarations for user data). This code
+is automatically inserted by the various transformations that are
+based on ``PSyDataTrans``, like ``ProfileTrans`` and
+``GOCeanExtractTrans``. More details can be found in
+:ref:`psy_data_transformation`. ::
+
+    USE extract_psy_data_mod, ONLY: extract_PSyDataType
+    TYPE(extract_PSyDataType), target, save :: extract_psy_data
+
+    CALL extract_psy_data%PreStart("update_field_mod", "update_field_code", 1, 1)
+    CALL extract_psy_data%PreDeclareVariable("a_fld", a_fld)
+    CALL extract_psy_data%PreDeclareVariable("b_fld", b_fld)
+    CALL extract_psy_data%PreEndDeclaration
+    CALL extract_psy_data%ProvideVariable("a_fld", a_fld)
+    CALL extract_psy_data%PreEnd
+
+    ! Begin of PSY-layer kernel call:
```

Comment:
I actually removed the comment, since it is not added to the code anymore, that was a left-over from when only comments were created for an extraction region.